rpc/server.go

```diff
@@ -79,12 +80,30 @@ func (rs *RpcServer) registerHandlers() error {
 				rs.client.Pay(obj.Channel, big.NewInt(int64(obj.Amount)))
 				return obj
 			})
+		case serde.GetPaymentChannelRequestMethod:
+			return processRequest(rs, requestData, func(r serde.GetPaymentChannelRequest) query.PaymentChannelInfo {
+				var pc query.PaymentChannelInfo
+				pc, err = rs.client.GetPaymentChannel(r.Id)
```

Comment:
```suggestion
				pc, err := rs.client.GetPaymentChannel(r.Id)
				// do something with the error
```

I find the syntax in the PR deceiving. `err` is assigned, but the `err` value is never checked since the assignment happens in a function that does not return an error. Our options are:
1. Modify [`RegisterRequestHandler`](https://github.com/statechannels/go-nitro/blob/1e95bec4a531ad81c2e86166220c01b61970d8bc/rpc/transport/transport.go#L31) to accept a function that can return an error. Something like:
``` go
RegisterRequestHandler(func([]byte) ([]byte, error)) error
```
2. Panic if `GetPaymentChannel` returns an error.

Reply:
> I find the syntax in the PR deceiving. err is assigned, but the err value is never checked since the assignment happens in a function that does not return an error.

Perhaps I'm mistaken but I think the `err` value is being checked. `err` is the return value defined [here](https://github.com/statechannels/go-nitro/blob/1e95bec4a531ad81c2e86166220c01b61970d8bc/rpc/server.go#L48). Inside the scope of `processRequest` we assign the `err` value to the result of `GetPaymentChannel`. The actual check happens just outside the scope of `processRequest`  where [we inspect the value of err](https://github.com/statechannels/go-nitro/blob/1e95bec4a531ad81c2e86166220c01b61970d8bc/rpc/server.go#L48).

I think in this case it's pretty confusing because in `processRequest` we both return something(`pc`) and assign a variable that lives outside the function scope (`err`). So we're probably better off doing something else.


> 1. Modify [`RegisterRequestHandler`](https://github.com/statechannels/go-nitro/blob/1e95bec4a531ad81c2e86166220c01b61970d8bc/rpc/transport/transport.go#L31) to accept a function that can return an error. Something like:
> 
This is probably the right thing to do, I was just being lazy and avoiding that.

>     2. Panic if `GetPaymentChannel` returns an error.

There's also a 3rd way where we `panic` but then `recover` the `panic` and return an error. [That was actually what I was doing previously to handle errors.](https://github.com/statechannels/go-nitro/pull/1206/commits/1e95bec4a531ad81c2e86166220c01b61970d8bc) It also makes me wonder if some of our functions should be using `recover` to guard against panics?

Reply:
My understanding of the function is:
1. Inside `registerHandlers`, there is a call to `RegisterRequestHandler` with a handler. That call can result in an error. That error is the return value of `registerHandlers`.
2. As of this PR, some of the handler callbacks can result in an error. Note that when an error like that is created, the `registerHandlers` has already returned. So the assignment has no effect.

The compiler tries points out to this issue if the syntax of error assignment is changed. Try:
``` go
return processRequest(rs, requestData, func(r serde.GetPaymentChannelRequest) query.PaymentChannelInfo {
  pc, err := rs.client.GetPaymentChannel(r.Id)
  return pc
})
```

The compiler should show: "err declared and not used". Another way to see this error is to use a different variable name like `err1`.

As far as panic/recover methodology: what are the advantages of this approach over the standard golang approach of returning and inspecting errors?

Reply:
>As of this PR, some of the handler callbacks can result in an error. Note that when an error like that is created, the registerHandlers has already returned. So the assignment has no effect.

This is the part I missed 🤦  I didn't pay attention that most of the code was a function declaration, not code being called immediately. 

> The compiler tries points out to this issue if the syntax of error assignment is changed. Try:

Just to be slightly nitpick/pedantic, but changing the `err` variable to a local declaration would show the `"err declared and not used"` error in a lot of scenarios where a valid assignment was being made. A function can set variables that are declared outside it's scope, and might not know how those variables are handled.

Here's a pretty trivial example.
```
func (rs *RpcServer) DoStuff() (err error) {
	func() {
        
		err = fmt.Errorf("BOOM")
                // This results in the error `err declared and not used`
                // err := fmt.Errorf("BOOM")
	}()

	return
}
```

Reply:
> My understanding of the function is:
> 
>     1. Inside `registerHandlers`, there is a call to `RegisterRequestHandler` with a handler. That call can result in an error. That error is the return value of `registerHandlers`.
> 
>     2. As of this PR, some of the handler callbacks can result in an error. Note that when an error like that is created, the `registerHandlers` has already returned. So the assignment has no effect.

Yup, I goofed here and forgot completely that these were callback handlers running after `registerHandlers` had been called. Sorry!

> The compiler tries points out to this issue if the syntax of error assignment is changed. Try:
> 
> ```go
> return processRequest(rs, requestData, func(r serde.GetPaymentChannelRequest) query.PaymentChannelInfo {
>   pc, err := rs.client.GetPaymentChannel(r.Id)
>   return pc
> })
> ```
> 
> The compiler should show: "err declared and not used". Another way to see this error is to use a different variable name like `err1`.
> 

Just to be slightly nitpicky/pedantic but switching `err` to a local declaration would show the `err declared and not used` error even in scenarios where the `err` value does get checked/used. Here's a trivial example:

```
func (rs *RpcServer) DoStuff() (err error) {
	func() {
		err = fmt.Errorf("BOOM")
		// This complains about err being unused
		// err := fmt.Errorf("BOOM")
	}()

	return
}
```

Reply:
For the code example above, if `err := ...` declares a new variable that is scoped to the anonymous function context. So the compiler error is useful. For example, the following test fails:

``` go
import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestError(t *testing.T) {
	assert.Error(t, ReturnsError())
}

func ReturnsError() (err error) {
	func() {
		err := fmt.Errorf("BOOM")
		fmt.Print(err)
	}()

	return
}
```

Reply:
For now I've updated it to just panic in https://github.com/statechannels/go-nitro/pull/1206/commits/63097d0f82af984ea599cfdd418f4b688a0efad9

Reply:
All that being said, the way `registerHandlers` is written (by me, I believe:)  ) lends itself to a confusing developer experience. So maybe we should refactor that function.

Reply:
> For the code example above, if `err := ...` declares a new variable that is scoped to the anonymous function context. So the compiler error is useful. For example, the following test fails:

I think I was just trying to clarify this comment in particular:

> The compiler tries points out to this issue if the syntax of error assignment is changed. Try:
> 
> ```go
> return processRequest(rs, requestData, func(r serde.GetPaymentChannelRequest) query.PaymentChannelInfo {
>   pc, err := rs.client.GetPaymentChannel(r.Id)
>   return pc
> })
> ```
> 
> The compiler should show: "err declared and not used". Another way to see this error is to use a different variable name like `err1`.

I just wanted to clarify that switching `err` to a local declaration and seeing it unused doesn't mean the previous `err` var wasn't being checked properly (although it wasn't).  It just proves that the `err` var isn't being checked in the local context.